File: backend/DTO/Request/OrderItemRequest.cs
using System.ComponentModel.DataAnnotations;

namespace backend.DTO.Request
{
    public class OrderItemRequest
    {
        [Required]
        public required int Quantity { get; set; }
        [Required]
        public required decimal Price { get; set; }
        [Required]
        public required int ProductId { get; set; }
        [Required]
        public required int OrderId { get; set; }
    }

}

File: frontend_admin/lib/model/request/order_item_request.dart
import 'package:equatable/equatable.dart';

class OrderItemRequest extends Equatable {
  const OrderItemRequest({this.quantity, this.price, this.productId, this.orderId});
  factory OrderItemRequest.fromJson(Map<String, dynamic> json) {
    return OrderItemRequest(
      quantity: json['quantity'],
      price: json['price']?.toDouble(),
      productId: json['productId'],
      orderId: json['orderId'],
    );
  }
  final int? quantity;
  final double? price;
  final int? productId;
  final int? orderId;

  Map<String, dynamic> toJson() {
    return {'quantity': quantity, 'price': price, 'productId': productId, 'orderId': orderId};
  }

  @override
  List<Object?> get props => [quantity, price, productId, orderId];
}
